apps/ltn/src/logic/turn_restrictions.rs

```diff
@@ -0,0 +1,347 @@
+/// "restricted_turns" = RoadB will be a member of "restricted_turns" if
+///                         a) RoadB is a member of "connected"
+///                         b) There is explicitly tagged turn restriction which prohibits traffic turning from
+///                            RoadA to RoadB, OR there is an explicitly tagged turn restriction which mandates
+///                            traffic from RoadA must turn onto a different road to RoadB.
+///                         c) RoadB is not a member of "opposing_oneways"
+/// possible_turns = These are turns that would be possible if all turn restrictions where removed.
+/// 
+/// Notes:
+/// -----
+/// * RoadA will NOT be a member of any of the groups connected, permitted, opposing_oneway, restricted_turn
+///   even if a no U-turns restriction exists
+/// * In reality a road/turn maybe signposted by both turn restrictions and oneway restrictions.
+///   Following (OSM practise)[https://wiki.openstreetmap.org/wiki/Relation:restriction#When_to_map] it is not 
+///   necessary mark turn restrictions when they are already implied by opposing oneway restrictions. We treat 
+///   "banned_turn" and "opposing_oneway" as mutually exclusive. 
+///
+/// Discouraged terms:
+/// -----------------
+/// "prohibited_turn" = use "restricted_turn" instead.
+/// "banned_turns" = use "restricted_turn" instead where practical. "Banned" is used elsewhere in A/BStreet,
+///                  (ie `RestrictionType::BanTurns`) but within the LTN tool "restricted" is preferred, as it
+///                  is more consistent with the `road.restricted_turns` and `road.complicated_turn_restrictions`
+///                  as well the general OSM tagging.
+/// "src_r" and "dst_r" = use "from_r" and "target_r" instead. ("src_r" and "dst_r" are too similar to
+///                       `road.src_i` and `road.dst_i` which are conceptually very different).
+pub struct FocusedTurns {
+    pub from_r: RoadID,
+    pub i: IntersectionID,
+    pub hull: Polygon,
+    pub possible_t: HashSet<RoadID>,
+    pub restricted_t: HashSet<RoadID>,
+}
+
+impl FocusedTurns {
+    pub fn new(r: RoadID, clicked_pt: Pt2D, map: &Map) -> Self {
+
+        let dst_i = map.get_r(r).dst_i;
+        let src_i = map.get_r(r).src_i;
+
+        let dst_m = clicked_pt.fast_dist(map.get_i(dst_i).polygon.center());
+        let src_m = clicked_pt.fast_dist(map.get_i(src_i).polygon.center());
+        
+        let i: IntersectionID;
+        if dst_m > src_m {
+            i = src_i;
+        } else {
+            i = dst_i;
+        }
+
+        let restricted_t = restricted_destination_roads(map, r, Some(i));
+        let possible_t = possible_destination_roads(map, r, Some(i));
+
+        let mut ft = FocusedTurns {
+            from_r: r,
+            i,
+            hull : Polygon::dummy(),
+            possible_t,
+            restricted_t,
+        };
+
+        ft.hull = hull_around_focused_turns(map, r, &ft.possible_t, &ft.restricted_t);
```

Comment:
Code style: it feels simpler to say `let hull = ....` above, using the two sets you define`. Then you can just implicitly return `FocusedTurns { ... }` and not declare anything

Reply:
Agreed - done